Source/UI/NewUI/PaymentForm/Factory/Theme/BillingForm/Theme+BillingCountryInput.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Theme+BillingCountryInput.swift
+//  
+//
+//  Created by Alex Ioja-Yang on 10/08/2022.
+//
+
+import UIKit
+
+public extension Theme {
+    
+    /// Theme generated Billing Country Input Style
+    struct ThemeBillingCountryInput: CellButtonStyle {
+        public var isMandatory: Bool = true
+        public var backgroundColor: UIColor = .clear
+        public var button: ElementButtonStyle
+        public var title: ElementStyle?
+        public var hint: ElementStyle?
+        public var mandatory: ElementStyle?
+        public var error: ElementErrorViewStyle?
+    }
+    
+    /// Create a Billing Country Input from provided themed components
+    func buildBillingCountryInput(button: CountryListButton,
+                                  title: ThemeTitle,
+                                  mandatory: ThemeMandatory? = nil,
+                                  subtitle: ThemeSubtitle? = nil,
+                                  error: ThemeError? = nil) -> ThemeBillingCountryInput {
+        ThemeBillingCountryInput(button: button,
+                                 title: title,
+                                 hint: subtitle,
+                                 mandatory: mandatory,
+                                 error: error)
+    }
+    
+    /// Create a Billing Country Input from provided content
+    func buildBillingCountryInput(buttonText: String,
+                                  buttonImage: UIImage? = nil,
+                                  title: String,
+                                  subtitle: String? = nil,
+                                  subtitleImage: UIImage? = nil,
+                                  isRequiredText: String? = nil,
+                                  errorText: String? = nil,
+                                  errorImage: UIImage? = nil) -> ThemeBillingCountryInput {
+        let subtitleText = subtitle ?? ""
+        let showSubtitle = !subtitleText.isEmpty || subtitleImage != nil
```

Comment:
for Cleaner Code, naming boolean to be `shouldSubtitleShow`

Reply:
I am not sure how that is Cleaner Code? Brevity is important, just adding `should` doesn't really improve anything?

Reply:
I guess may be readability ? . Also I was looking at this article here https://betterprogramming.pub/naming-conventions-in-swift-4b7ca5eed4d2. May be not a good example. But definitely what we need is code convention guidelines which we can follow for future.

Reply:
[Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines) should be the guidelines we follow, they are hosted on the official Swift open source page?

They're not overbearing which is why we're probably having this discussion? 

If we look at 
> Naming
Promote Clear Usage

there's a line of 
> Omit needless words. Every word in a name should convey salient information at the use site.

which imo explains why "just adding should" doesn't make it better. 

***

>Name variables, parameters, and associated types according to their roles, rather than their type constraints.

In my understanding "showSubtitle" meets very well this naming as its role is to decide "if subtitle will be shown" whilst not carrying any extra words that don't convey extra information.

Reply:
See other comment on [how syntax came to be](https://github.com/checkout/frames-ios/pull/245#discussion_r944429957) and why in my opinion it is perfectly valid readable syntax for Swift consumer.

Looking at some WWDC talks this format will again become very familiar very quickly!